zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/GroupController.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller.usermanagement;
+
+import io.camunda.identity.usermanagement.CamundaGroup;
+import io.camunda.identity.usermanagement.CamundaUser;
+import io.camunda.identity.usermanagement.service.GroupService;
+import io.camunda.identity.usermanagement.service.MembershipService;
+import io.camunda.zeebe.gateway.protocol.rest.SearchRequestDto;
+import io.camunda.zeebe.gateway.rest.controller.ZeebeRestController;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.dto.AssignUserToGroupRequest;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.dto.SearchResponseDto;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ZeebeRestController
+@RequestMapping("/v2")
+public class GroupController {
+  private final GroupService groupService;
+  private final MembershipService membershipService;
+
+  public GroupController(
+      final GroupService groupService, final MembershipService membershipService) {
+    this.groupService = groupService;
+    this.membershipService = membershipService;
+  }
+
+  @PostMapping(
+      path = "/groups",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.CREATED)
+  public CamundaGroup createGroup(@RequestBody final CamundaGroup group) {
+    return groupService.createGroup(group);
+  }
+
+  @DeleteMapping(path = "/groups/{id}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void deleteGroup(@PathVariable(name = "id") final Long groupId) {
```

Comment:
❓ Is there a reason to not just use `id` as a variable name? This will remove the need to specify path variable name

```java
  public void deleteGroup(@PathVariable final Long id) {
```

Reply:
This also applies to other endpoints

Reply:
I've tried to stay explicit and consistent in a way to always have fully meaningful variable names in the method bodies. In other methods there for example, we have two IDs in the body (e.g. `groupId` and `userId` ... this way, when in the method, you always know which IDs you're dealing with. Changing `groupId` to `id` only would then always let us make that extra step: "`id` ... uhm, group, user, role, permission ... let me double check the path on top" ... thus the very explicit variable naming.

I think despite the few extra bits mapping the path variable in `.../{id}` to `groupId` in Java it overall helps when developing.